src/components/GraphFilter/GraphSettings.tsx

```diff
@@ -54,6 +54,18 @@ class GraphSettings extends React.PureComponent<GraphSettingsProps> {
     }
   }
 
+  private onToggle = isOpen => {
+    this.setState({
+      isOpen
+    });
+  };
+
+  private onSelect = _event => {
+    this.setState({
```

Comment:
you can just remove this `setState` here to avoid what @jshaughn mentioned, ie. dropdown being closed on first click